tests/test_daemon.cpp

```diff
@@ -375,6 +378,18 @@ struct DaemonCreateLaunchAliasTestSuite : public Daemon, public FakeAliasConfig,
     {
         EXPECT_CALL(mpt::MockStandardPaths::mock_instance(), writableLocation(_))
             .WillRepeatedly(Return(fake_alias_dir.path()));
+
+        EXPECT_CALL(mock_json_utils, write_json(_, _)).WillRepeatedly([](auto&& a, auto&& b) {
+            // this try catch is necessary since propagating the error hangs the test.
+            try
+            {
+                MP_JSONUTILS.JsonUtils::write_json(a, b);
```

Comment:
could use perfect forwarding here:

```suggestion
                MP_JSONUTILS.JsonUtils::write_json(std::forward<decltype(a)>(a), std::forward<decltype(b)>(b));
```